db/models/articles.models.js

```diff
@@ -55,11 +55,41 @@ const updateArticleVotes = (articleId, incVotes) => {
     .then(result => result.rows[0]);
 };
 
+const fetchAllArticles = (sort_by = 'created_at', order = 'desc', topic) => {
+  const validSortColumns = ['created_at', 'title', 'author', 'votes', 'article_id'];
+  const validOrders = ['asc', 'desc'];
+
+  if (!validSortColumns.includes(sort_by)) {
+      return Promise.reject({ status: 400, msg: 'Invalid sort_by column' });
+  }
+
+  if (!validOrders.includes(order)) {
+      return Promise.reject({ status: 400, msg: 'Invalid order query' });
+  }
+
+  let queryStr = `
+      SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+      FROM articles
+      LEFT JOIN comments ON articles.article_id = comments.article_id
+  `;
+
+  if (topic) {
```

Comment:
There should be a way of checking if the topic exists so we know the reason why if we get an empty array back (The existence of the topic should come from the db, not to be hard coded so it remains up to date)